File: beginner_tutorials/scripts/listener_msg_sub.py
#!/usr/bin/python3
import rospy
from beginner_tutorials.msg import Pubsub # 作成したメッセージをインポート


# Subscribeした際のコールバック関数
def callback(data):
  rospy.loginfo(rospy.get_caller_id() + 'I heard python %s', data.msg + ": " + str(data.count))


# listener関数
def listener():
  rospy.init_node('listener_msg_py', anonymous=True)

  # Subscriberの設定(メッセージとしてPubsubを使う)
  rospy.Subscriber('chatter_msg_py', Pubsub, callback)

  # ROSの無限ループ待ち
  rospy.spin()


if __name__ == '__main__':
  listener()

File: beginner_tutorials/src/listener_msg_sub.cpp
#include "ros/ros.h"
#include "beginner_tutorials/Pubsub.h" // 作成したメッセージ(パッケージ名/メッセージ名.hの形でinclude)


// Subscribeした際のコールバック関数(メッセージのPubsubを使用)
void chatterCallback(const beginner_tutorials::Pubsub::ConstPtr& data)
{
  // 表示用の文字列作成
  std::stringstream ss;
  ss << data->msg << ": " << data->count;
  char char_array[256];
  ss.get(char_array, 256);

  ROS_INFO("I heard: [%s]", char_array); // ROSのログにINFOレベルで書き込み(表示)
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "listener_msg");
  ros::NodeHandle n;

  // Subscriberの設定。
  ros::Subscriber sub = n.subscribe("chatter_msg", 1000, chatterCallback);

  // ROSの無限ループ待ち
  ros::spin();

  return 0;
}